ambari-server/src/main/java/org/apache/ambari/server/topology/ConfigurableHelper.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ambari.server.topology;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.apache.ambari.server.controller.internal.BlueprintResourceProvider.PROPERTIES_ATTRIBUTES_PROPERTY_ID;
+import static org.apache.ambari.server.controller.internal.BlueprintResourceProvider.PROPERTIES_PROPERTY_ID;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+
+/**
+ * Helper class for {@link Configurable}
+ */
+public class ConfigurableHelper {
+
+  static final ImmutableSet<String> PERMITTED_CONFIG_FIELDS = ImmutableSet.of(PROPERTIES_PROPERTY_ID, PROPERTIES_ATTRIBUTES_PROPERTY_ID);
+
+  /**
+   * Parses configuration maps The configs can be in fully structured JSON, e.g.
+   * <code>
+   * [{"hdfs-site":
+   *  "properties": {
+   *    ""dfs.replication": "3",
+   *    ...
+   *  },
+   *  properties_attributes: {}
+   * }]
+   * </code>
+   * or flattened like
+   * <code>
+   * [{
+   *  "hdfs-site/properties/dfs.replication": "3",
+   *  ...
+   * }]
+   * </code>
+   * In the latter case it calls {@link ConfigurationFactory#getConfiguration(Collection)}
+   * @param configs
+   * @return
+   */
+  public static Configuration parseConfigs(@Nullable Collection<? extends Map<String, ?>> configs) {
+    Configuration configuration;
+
+    if (null == configs) {
```

Comment:
Adding `|| configs.isEmpty()` would make it easier to see what's going to happen in that case.